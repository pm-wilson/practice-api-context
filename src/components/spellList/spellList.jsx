import React from 'react';
import { useSpells } from '../../hooks/spells';
import { useColorPicker } from '../../hooks/colors';

const SpellList = () => {
  const { spellList } = useSpells();
  const { lightMode } = useColorPicker();
  const modeClass = lightMode ? '_1yf-o0z033RbAHQOAY1KRe' : '_2hAGQDxUbRuNG6XUTI7sOp';

  if(spellList.length === 0) return <h1>No Data</h1>;
  const spellListToShow = spellList.results;

  const spellItems = spellListToShow.map(spell => (<div key={spell.index}>{spell.name}</div>));

  return (
    <div className={modeClass}>
      <h2>Spell List</h2>
      <div>
        {spellItems}
      </div>
    </div>
  );
};

export default SpellList;
