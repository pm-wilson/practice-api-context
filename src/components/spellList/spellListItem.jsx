import React from 'react';
import PropTypes from 'prop-types';

const SpellListItem = ({ name }) => (
  <h6>{name}</h6>
);

SpellListItem.propTypes = {
  name: PropTypes.string.isRequired,
};

export default SpellListItem;
