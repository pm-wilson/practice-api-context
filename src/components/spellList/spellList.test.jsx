import React from 'react';
import { render } from '@testing-library/react';
import spellList from './spellList';
import { MemoryRouter } from 'react-router-dom';

describe('spellList component', () => {
  it('renders spellList', () => {
    const { asFragment } = render(<MemoryRouter><spellList /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
