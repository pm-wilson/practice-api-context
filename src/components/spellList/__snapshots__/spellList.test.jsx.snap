// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spellList component renders spellList 1`] = `
<DocumentFragment>
  <spelllist />
</DocumentFragment>
`;
